File: src/main/java/com/nicholasnassar/sfsapi/ApacheCompletableFuture.java
package com.nicholasnassar.sfsapi;

import org.apache.http.concurrent.FutureCallback;

import java.util.concurrent.CompletableFuture;

public class ApacheCompletableFuture<T> extends CompletableFuture<T> implements FutureCallback<T> {
    @Override
    public void completed(T result) {
        complete(result);
    }

    @Override
    public void failed(Exception ex) {
        ex.printStackTrace();

        completeExceptionally(ex);
    }

    @Override
    public void cancelled() {
        //TODO: Hopefully does not occur
    }
}

File: src/main/java/com/nicholasnassar/sfsapi/ApacheCompletableFuture.kt
package com.nicholasnassar.sfsapi

import org.apache.http.concurrent.FutureCallback
import java.lang.Exception
import java.util.concurrent.CompletableFuture

class ApacheCompletableFuture<T> : CompletableFuture<T>(), FutureCallback<T> {
    override fun completed(result: T) {
        complete(result)
    }

    override fun failed(ex: Exception) {
        ex.printStackTrace()

        completeExceptionally(ex)
    }

    override fun cancelled() {
        //TODO: Hopefully does not occur
    }
}